src/test/java/core/basesyntax/RegistrationServiceTests.java

```diff
@@ -0,0 +1,147 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class RegistrationServiceTests {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeAll
+    public static void set_up() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    public void register_validUser_success() {
+        User newUser = new User();
+        newUser.setLogin("Bob679");
+        newUser.setAge(18);
+        newUser.setPassword("0987654321");
+        registrationService.register(newUser);
+        User expectedUser = storageDao.get(newUser.getLogin());
+        assertEquals(expectedUser.getLogin(), newUser.getLogin());
+        assertEquals(expectedUser.getPassword(), newUser.getPassword());
+        assertEquals(expectedUser.getAge(), newUser.getAge());
+        assertEquals(storageDao.size(), 1);
+    }
+
+    @Test
+    public void register_incorrectLogin_throwsException() {
+        User newUser = new User();
+        newUser.setLogin("Mary");
+        newUser.setAge(18);
+        newUser.setPassword("666666666");
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("The length of login is less than 6 characters", exception.getMessage());
+    }
+
+    @Test
+    public void register_existingLogin_throwsException() {
+        User newUser = new User();
+        newUser.setLogin("Bob679");
+        newUser.setAge(22);
+        newUser.setPassword("1230988686");
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("The user with the same login already exists", exception.getMessage());
```

Comment:
This test checks for an existing login, but there is no setup to ensure that a user with the login 'Bob679' already exists in the storage. Use low-level methods to add a user to the storage before running this test.